backends/occa/ceed-occa.c

```diff
@@ -16,6 +16,45 @@
 #define CEED_DEBUG_COLOR 10
 #include "ceed-occa.h"
 
+// *****************************************************************************
+// * CeedDladdr_Occa for Apple and Linux
+// *****************************************************************************
+#if defined(__APPLE__) || defined(__linux__)
+static int CeedDladdr_Occa(Ceed ceed) {   
+  Dl_info info;
+  Ceed_Occa *data = ceed->data;
+  memset(&info,0,sizeof(info));
+  const int ierr = dladdr((void*)&CeedInit,&info);
+  dbg("[CeedDladdr_Occa] found %s", info.dli_fname);
+  if (ierr==0)
+    return CeedError(ceed, 1, "OCCA backend cannot fetch dladdr");
+  // libceed_dir setup & alloc in our data
+  const char *libceed_dir = (char*)info.dli_fname;
+  const int path_len = strlen(libceed_dir);
+  dbg("[CeedDladdr_Occa] path_len=%d", path_len);
+  CeedCalloc(path_len+1,&data->libceed_dir);
+  // stat'ing the library to make sure we can access the path
+  struct stat buf;
+  if (stat(libceed_dir, &buf)!=0)
+    return CeedError(ceed, 1, "OCCA backend cannot stat %s", libceed_dir);
+  // Now remove libceed.so part to get the path
+  char *last_slash = strrchr(libceed_dir,'/');
+  if (!last_slash)
+    return CeedError(ceed, 1,
+                     "OCCA backend cannot locate libceed_dir from %s",
+                     libceed_dir);
+  // remove from last_slash and push it to our data
+  *last_slash=0;
```

Comment:
We aren't supposed to modify `libceed_dir`, which is `const`. I think you can just 

    memcpy(data->libceed_dir,libceed_dir,last_slash-libceed_dir);

which will be null-terminated because `CeedCalloc` returns zeroed memory.

Reply:
outdated with the refactoring